### PR TITLE
Add push notifications (only if a tab is open in browser)

### PR DESCRIPTION
**Pull request by Ksinia**

### CODE REVIEW

File: src/notificationsMiddleware.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+type Options = {
+  title: string;
+  subtitle?: string;
+  message?: string;
+  duration?: number;
+  icon?: string;
+  vibrate?: number | number[];
+  silent?: boolean;
+  onClick?: onClickType;
+  requireInteraction?: boolean;
```
Comment (paskal) [inline src/notificationsMiddleware.ts:15]: I propose to remove unused options here.

File: src/notificationsMiddleware.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    setTimeout(not.close.bind(not), duration || defaultDuration);
+  }
+};
+
+const notificationsMiddleware = (store: any) => (next: Dispatch) => (
+  action: AnyAction
+) => {
+  if (
+    store.getState().notificationsSupported &&
+    action.type === PUSH_NOTIFICATION
```
Comment (paskal) [inline src/notificationsMiddleware.ts:59]: I propose to move this up.

**Review by paskal — APPROVED** [review]
LGTM except two small changes.

File: src/notificationsMiddleware.ts (hunk 1 above)
Comment (Ksinia) [reply src/notificationsMiddleware.ts:15]: removed

**Pull request merged**